kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/Patchable.java

```diff
@@ -15,8 +15,35 @@
  */
 package io.fabric8.kubernetes.client.dsl;
 
+import io.fabric8.kubernetes.client.dsl.base.PatchContext;
+
 public interface Patchable<T> {
 
+  /**
+   * Update field(s) of a resource using strategic a JSON patch.
+   *
+   * @param item item to be patched with patched values
+   * @return returns deserialized version of api server response
+   */
   T patch(T item);
 
+  /**
+   * Update field(s) of a resource using strategic merge patch or a JSON patch.
+   *
+   * @param patch The patch to be applied to the resource JSON file.
+   * @return returns deserialized version of api server response
+   */
+  default T patch(String patch) {
+    return patch(null, patch);
+  }
+
+  /**
+   * Update field(s) of a resource using strategic merge patch or a JSON patch.
```

Comment:
Is it always using a strategic merge patch, or does it depend on the provided patchContext?